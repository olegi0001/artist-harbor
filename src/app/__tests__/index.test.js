import React from 'react'
import { create, act } from 'react-test-renderer'
import SearchScreen from '../index'

// Mock the API requests module
jest.mock('../../api/requests', () => ({
	searchArtist: jest.fn()
}))

beforeEach(() => {
	jest.clearAllMocks()
})

describe('SearchScreen', () => {
	it('renders correctly', () => {
		const tree = create(<SearchScreen />)
		expect(tree.toJSON()).toMatchSnapshot()
	})
})

it('updates searchQuery state when input changes', () => {
	const component = create(<SearchScreen />)
	const input = component.root.findByType('TextInput')

	act(() => {
		input.props.onChangeText('test')
	})

	expect(component.root.findByType('TextInput').props.value).toBe('test')
})

it('fetches search result on valid search query', async () => {
	const component = create(<SearchScreen />)
	const input = component.root.findByType('TextInput')

	// Simulate typing
	act(() => {
		input.props.onChangeText('randomQuery')
	})

	// Trigger useEffect
	await act(async () => {
		await new Promise((resolve) => setTimeout(resolve, 300))
	})

	// Ensure searchArtist function is called with the correct argument
	expect(require('../../api/requests').searchArtist).toHaveBeenCalledWith('randomQuery')
})

it('does not fetch search result on invalid search query', async () => {
	const component = create(<SearchScreen />)
	const input = component.root.findByType('TextInput')

	// Simulate typing
	act(() => {
		input.props.onChangeText('q')
	})

	// Trigger useEffect
	await act(async () => {
		await new Promise((resolve) => setTimeout(resolve, 300))
	})

	// Ensure searchArtist function is not called
	expect(require('../../api/requests').searchArtist).not.toHaveBeenCalled()
})

it('displays loading indicator when loading state is true', async () => {
	const component = create(<SearchScreen />)
	const input = component.root.findByType('TextInput')

	// Simulate typing
	act(() => {
		input.props.onChangeText('randomQueryForLoading')
	})

	// Trigger useEffect
	await act(async () => {
		await new Promise((resolve) => setTimeout(resolve, 300))
	})

	expect(component.root.findByType('ActivityIndicator')).toBeTruthy()
})
