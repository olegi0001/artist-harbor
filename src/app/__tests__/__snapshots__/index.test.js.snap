// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchScreen renders correctly 1`] = `
[
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      {
        "backgroundColor": "#f0b9ef",
        "flex": 0,
      }
    }
  />,
  <ViewManagerAdapter_ExpoLinearGradient
    colors={
      [
        4293966319,
        4289647087,
      ]
    }
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "alignSelf": "stretch",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "flex": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingHorizontal": 30,
            "paddingTop": 30,
            "width": "100%",
          }
        }
      >
        <Text />
        <TextInput
          onChangeText={[Function]}
          placeholder="Search for artists"
          placeholderTextColor="gray"
          style={
            {
              "borderColor": "gray",
              "borderRadius": 16,
              "borderWidth": 1,
              "color": "gray",
              "flex": 1,
              "height": 40,
              "paddingHorizontal": 16,
              "width": "100%",
            }
          }
          value=""
        />
      </View>
      <View
        style={
          {
            "height": "100%",
            "paddingHorizontal": 60,
          }
        }
      >
        <ActivityIndicator
          animating={false}
          color="gray"
          hidesWhenStopped={true}
          size="small"
          style={
            {
              "marginLeft": 8,
            }
          }
        />
      </View>
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>,
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      {
        "backgroundColor": "#aed1ef",
        "flex": 0,
      }
    }
  />,
]
`;
